src/messages/components/AttachmentGallery.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <Teleport to="body">
+    <div class="gallery">
+      <QCarousel
+        ref="carouselRef"
+        v-model="currentId"
+        transition-prev="slide-right"
+        transition-next="slide-left"
+        :arrows="moreThanOneImage"
+        :swipeable="moreThanOneImage"
+        :thumbnails="moreThanOneImage"
+        animated
+        control-color="white"
+        infinite
+        padding
+        :class="Platform.is.mobile ? '' : 'q-pa-xl'"
+        @click="event => closeIfOverlay(event)"
+      >
+        <QCarouselSlide
+          v-for="image in images"
+          :key="image.id"
+          :name="image.id"
+          :img-src="image.urls.preview || image.urls.original"
+        />
+        <template #control>
+          <QCarouselControl
+            position="top-right"
+            :offset="[12, 12]"
+          >
+            <QBtn
+              icon="fas fa-download"
+              rounded
+              color="white"
+              text-color="primary"
+              :href="currentAttachment?.urls?.download"
+              :label="$t('BUTTON.DOWNLOAD')"
+            />
+            <QBtn
+              rounded
+              icon="fas fa-times"
+              color="white"
+              text-color="primary"
+              class="q-ml-sm"
+              :label="$t('BUTTON.CLOSE')"
+              @click="emit('close')"
+            />
+          </QCarouselControl>
+        </template>
+      </QCarousel>
+    </div>
+  </Teleport>
+</template>
+
+<script setup>
+import {
+  QCarousel,
+  QCarouselSlide,
+  QCarouselControl,
+  QBtn,
+  Platform,
+} from 'quasar'
+import { computed, ref } from 'vue'
+import { useEvent } from 'vue-composable'
```

Comment:
Could use `import { useEventListener } from '@vueuse/core'` instead, as we already have vueuse.

Reply:
Seems good to avoid another dependency.